### PR TITLE
[#HW-2] - blog basic layout

### PR DESCRIPTION
**Pull request by Oleksandr96**

### CODE REVIEW

**Comment by Oleksandr96** [comment]
- Basic layout and components;
- Post creation form, form validation
- Service for posts

**Comment by Oleksandr96** [comment]
- Basic layout;
- Post creation form and validation
- Posts service

image

image

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
 })
+
 export class AppComponent {
   title = 'bootcamp-blog';
+
+  posts = []
+
+
+  constructor(private appPostsService: AppPostsService) {
+    // @ts-ignore
```
Comment (MMaksymivUa) [inline src/app/app.component.ts:18]: do not do this

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
+
 export class AppComponent {
   title = 'bootcamp-blog';
+
+  posts = []
```
Comment (MMaksymivUa) [inline src/app/app.component.ts:14]: types

File: src/app/app.module.ts [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
+import { ModalComponent } from './components/modal/modal.component';
+import { PostFormComponent } from './components/post-form/post-form.component';
+import {AppPostsService} from "./services/app-posts.service";
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    HeaderComponent,
+    PostComponent,
```
Comment (MMaksymivUa) [inline src/app/app.module.ts:17]: create feature Posts module

File: src/app/components/header/header.component.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
+header{
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+  align-items: center;
+  padding: 20px 10px;
+}
+.menu {
+  &__item{
+    color: #bababa;
```
Comment (MMaksymivUa) [inline src/app/components/header/header.component.scss:10]: variable

File: src/app/components/modal/modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+@Component({
+  selector: 'app-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss']
+})
+export class ModalComponent implements OnInit {
+  @Input() title = 'Add new article'
+  @Output() close = new EventEmitter<void>()
+
+
```
Comment (MMaksymivUa) [inline src/app/components/modal/modal.component.ts:12]: use linter

File: src/app/components/post-form/post-form.component.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  <div class="form__item">
+    <label>Text</label>
+    <textarea formControlName="text" id="text" name="text" ></textarea>
+    <div
+      *ngIf="newPostForm.get('text')!.invalid && newPostForm.get('text')!.touched"
+      class="validation">
+
+      <span *ngIf="newPostForm.get('text')!.errors?.['required']" >Field required</span>
+      <span *ngIf="newPostForm.get('text')!.errors?.['minlength']" >
+        The minimum field length is 30 characters. Current length is {{newPostForm.get('text')!.errors?.['minlength']['actualLength']}}
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.html:50]: let's create a map of errors in class, like error map for elements <ErrorKey, ErrorMessage> and an error label one error element in HTML

File: src/app/components/post-form/post-form.component.scss [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    width: 100%;
+    margin-bottom: 15px;
+    label{
+      display: block;
+      margin: 5px 0 5px 0;
+      font-weight: bold;
+    }
+    input, textarea{
+      width: 100%;
+      max-width: 100%;
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.scss:23]: no need

File: src/app/components/post-form/post-form.component.scss [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  &.ng-invalid.ng-touched{
+    border: 2px solid #d00f0f;
+  }
+  &.ng-valid.ng-touched{
+    border:2px solid #60b65e;
+  }
+}
+.form{
+  width: 500px;
+  display: block;
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.scss:11]: but it is a block

File: src/app/components/post-form/post-form.component.scss [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  &__item{
+    display: block;
+    width: 100%;
+    margin-bottom: 15px;
+    label{
+      display: block;
+      margin: 5px 0 5px 0;
```
Comment (MMaksymivUa) [inline src/app/components/post-form/post-form.component.scss:18]: you can write it simpler: margin: 5px 0;

File: src/app/components/post/post.component.html [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<article class="post">
```
Comment (MMaksymivUa) [inline src/app/components/post/post.component.html:1]: check if post exist

File: src/app/components/post/post.component.scss [hunk 11]
```diff
@@ -0,0 +1,60 @@
+.post{
```
Comment (MMaksymivUa) [inline src/app/components/post/post.component.scss:1]: use code formatting tools

File: src/app/components/sidebar/sidebar.component.scss [hunk 12]
```diff
@@ -0,0 +1,9 @@
+.sidebar{
+  color: #555458;
+  background-color: #f3f3f3;
```
Comment (MMaksymivUa) [inline src/app/components/sidebar/sidebar.component.scss:3]: variables

File: src/app/services/app-posts.service.ts [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  title: string,
+  author: string,
+  text: string,
+  date: Date,
+  likes: number
+}
+@Injectable({
+  providedIn: 'root'
+})
+
```
Comment (MMaksymivUa) [inline src/app/services/app-posts.service.ts:14]: baaaaad formatting

File: src/app/services/app-posts.service.ts [hunk 14]
```diff
@@ -0,0 +1,44 @@
+import {Injectable} from "@angular/core";
+
+export interface Post {
+  id?: number,
+  title: string,
+  author: string,
+  text: string,
+  date: Date,
+  likes: number
+}
```
Comment (MMaksymivUa) [inline src/app/services/app-posts.service.ts:10]: move to model.file

File: src/app/services/app-posts.service.ts (hunk 13 above)
Comment (Oleksandr96) [reply src/app/services/app-posts.service.ts]: fixed

File: src/app/components/post/post.component.scss (hunk 11 above)
Comment (Oleksandr96) [reply src/app/components/post/post.component.scss]: fixed

File: src/app/components/modal/modal.component.ts (hunk 5 above)
Comment (Oleksandr96) [reply src/app/components/modal/modal.component.ts]: fixed

File: src/app/app.component.ts (hunk 2 above)
Comment (Oleksandr96) [reply src/app/app.component.ts]: fixed

File: src/app/app.component.ts (hunk 1 above)
Comment (Oleksandr96) [reply src/app/app.component.ts]: fixed

File: src/app/services/app-posts.service.ts (hunk 14 above)
Comment (Oleksandr96) [reply src/app/services/app-posts.service.ts]: fixed: replace to /interfaces/post.interface.ts

File: src/app/components/post/post.component.html (hunk 10 above)
Comment (Oleksandr96) [reply src/app/components/post/post.component.html]: fixed

File: src/app/components/post-form/post-form.component.scss (hunk 8 above)
Comment (Oleksandr96) [reply src/app/components/post-form/post-form.component.scss:13]: fixed

File: src/app/components/post-form/post-form.component.scss (hunk 7 above)
Comment (Oleksandr96) [reply src/app/components/post-form/post-form.component.scss:28]: fixed

File: src/app/app.module.ts (hunk 3 above)
Comment (Oleksandr96) [reply src/app/app.module.ts:17]: fixed

File: src/app/components/header/header.component.scss (hunk 4 above)
Comment (Oleksandr96) [reply src/app/components/header/header.component.scss:11]: fixed

File: src/app/components/sidebar/sidebar.component.scss (hunk 12 above)
Comment (Oleksandr96) [reply src/app/components/sidebar/sidebar.component.scss:3]: fixed

File: src/app/components/post-form/post-form.component.html (hunk 6 above)
Comment (Oleksandr96) [reply src/app/components/post-form/post-form.component.html]: fixed

**Pull request merged**